addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1933,23 +1953,27 @@ var FieldProgressBar = AbstractField.extend({
         }
         return this._super();
     },
+    /**
+     * Updates the widget with value
+     *
+     * @param {Number} value
+     */
     on_update: function (value) {
         if (!isNaN(value)) {
             if (this.edit_max_value) {
-                try {
-                    this.max_value = this._parseValue(value);
-                    this._isValid = true;
-                } catch (e) {
-                    this._isValid = false;
-                }
+                this.max_value = value;
+                this._isValid = true;
                 var changes = {};
                 changes[this.nodeOptions.max_value] = this.max_value;
                 this.trigger_up('field_changed', {
                     dataPointID: this.dataPointID,
                     changes: changes,
                 });
             } else {
-                this._setValue(value);
+                // _setValues accepts string and will parse it
+                var formattedValue = this._formatValue(value);
+                this._setValue(formattedValue);
+                this.value = value;
```

Comment:
theoretically, this will be done by the reset (following the _setValue), so you don't have to do it yourself